src/frontend/src/pages/PIADetailPage/index.tsx

```diff
@@ -104,50 +105,78 @@ const PIADetailPage = () => {
     setDownloadError('');
   };
 
+  const handleShowModal = (modalType: string) => {
+    switch (modalType) {
+      case 'submit':
+        setModalConfirmLabel(messages.Modal.Submit.ConfirmLabel.en);
+        setModalCancelLabel(messages.Modal.Submit.CancelLabel.en);
+        setModalTitleText(messages.Modal.Submit.TitleText.en);
+        setModalParagraph(messages.Modal.Submit.ParagraphText.en);
+        setModalButtonValue('submit');
+        break;
+      case 'edit':
+        setModalConfirmLabel(messages.Modal.Edit.ConfirmLabel.en);
+        setModalCancelLabel(messages.Modal.Edit.CancelLabel.en);
+        setModalTitleText(messages.Modal.Edit.TitleText.en);
+        setModalParagraph(messages.Modal.Edit.ParagraphText.en);
+        break;
```

Comment:
To make ensure that we don't accidentally send in 'submit' as a value when it isn't what we want.
```suggestion
        setModalButtonValue('edit');
        break;
```